source/app-development/add-jupyter/modify-submit-parameters.rst

```diff
@@ -151,6 +151,85 @@ For Torque, you can choose processors-per-node with:
    returned from web form for "Number of nodes". We check if it is blank and
    return a valid number (since it wouldn't make sense to return ``0``).
 
+PBS Professional
+````````````````
+
+For most cases of PBS Professional you will want to modify how the
+``bc_num_slots`` (number of CPUs on a single node) is submitted to the batch
+server.
+
+This can be specified as such:
+
+.. code-block:: yaml
+
+   # submit.yml.erb
+   ---
+
+   batch_connect
+     template: "basic"
+
+   script:
+     native: [ "-l", "select=1:ncpus=<%= bc_num_slots.blank? ? 1 : bc_num_slots.to_i %>" ]
+
+where we define the :command:`qsub` parameters as an array under ``script`` and
+``native``.
+
+If you would like to mimic how Torque handles ``bc_num_slots`` (number of
+**nodes**), then we will first need to change the form label of
+``bc_num_slots`` that the user sees in the form. This can be done by adding to
+the form configuration file the following option:
+
+.. code-block:: yaml
+
+   # form.yml
+   ---
+
+   ...
+   attributes:
+
+     ...
+
+     bc_num_slots:
+       label: "Number of nodes"
```

Comment:
Use a diff and show the actual yaml file sans comments: (one or both of these)

From:

```yaml
---
cluster: ""

attributes:
  modules: "python"
  conda_extensions: "1"
  extra_jupyter_args: ""

form:
  - modules
  - conda_extensions
  - extra_jupyter_args
  - bc_num_hours
  - bc_num_slots
  - bc_account
  - bc_queue
  - bc_email_on_started
```

To:

```diff
---
cluster: ""

attributes:
  modules: "python"
  conda_extensions: "1"
  extra_jupyter_args: ""
+ bc_num_slots:
+   label: "Number of nodes"

form:
  - modules
  - conda_extensions
  - extra_jupyter_args
  - bc_num_hours
  - bc_num_slots
  - bc_account
  - bc_queue
  - bc_email_on_started
```